src/main/java/org/jabref/gui/entryeditor/RelatedArticlesTab.java

```diff
@@ -72,8 +75,10 @@ private StackPane getRelatedArticlesPane(BibEntry entry) {
         ProgressIndicator progress = new ProgressIndicator();
         progress.setMaxSize(100, 100);
 
-        MrDLibFetcher fetcher = new MrDLibFetcher(preferencesService.getWorkspacePreferences().getLanguage().name(),
-                Globals.BUILD_INFO.version, preferencesService.getMrDlibPreferences());
+        MrDLibFetcher fetcher = new MrDLibFetcher(
+                preferencesService.getWorkspacePreferences().getLanguage().name(),
+                buildInfo.version,
```

Comment:
Why not injecting here?

```java
Injector.instantiateModelOrService(BuildInfo.class).version
```

Or even in the fetcher itself?

Reply:
Could be. I opted for constructor injection here and recieve the build info from the entry editor view class, as afterburner is not a dependency injection framework, but an fxml mvvm framework providing a presenterfactory that injects into the entry editor view the class.

Implementing a proper DI framework would be the next step.